cluster-api/src/main/java/io/aiven/klaw/clusterapi/controller/ClusterApiController.java

```diff
@@ -0,0 +1,252 @@
+
+  SchemaService schemaService;
+
+  MonitoringService monitoringService;
+
+  AivenApiService aivenApiService;
+
+  @RequestMapping(
+      value = "/getApiStatus",
+      method = RequestMethod.GET,
+      produces = {MediaType.APPLICATION_JSON_VALUE})
+  public ResponseEntity<ClusterStatus> getApiStatus() {
+    return new ResponseEntity<>(ClusterStatus.ONLINE, HttpStatus.OK);
+  }
+
+  //    @RequestMapping(value = "/reloadTruststore/{protocol}/{clusterName}", method =
+  // RequestMethod.GET,produces = {MediaType.APPLICATION_JSON_VALUE})
+  //    public ResponseEntity<String> reloadTruststore(@PathVariable String protocol,
+  //                                                   @PathVariable String clusterName){
+  //        return new ResponseEntity<>(manageKafkaComponents.reloadTruststore(protocol,
+  // clusterName), HttpStatus.OK);
+  //    }
+
+  @RequestMapping(
+      value = "/getStatus/{bootstrapServers}/{protocol}/{clusterName}/{clusterType}",
+      method = RequestMethod.GET,
+      produces = {MediaType.APPLICATION_JSON_VALUE})
+  public ResponseEntity<ClusterStatus> getStatus(
+      @PathVariable String bootstrapServers,
+      @Valid @PathVariable KafkaSupportedProtocol protocol,
+      @PathVariable String clusterName,
+      @PathVariable String clusterType) {
+    return new ResponseEntity<>(
+        utilComponentsService.getStatus(bootstrapServers, protocol, clusterName, clusterType),
+        HttpStatus.OK);
+  }
+
+  @RequestMapping(
+      value = "/getTopics/{bootstrapServers}/{protocol}/{clusterName}",
+      method = RequestMethod.GET,
+      produces = {MediaType.APPLICATION_JSON_VALUE})
+  public ResponseEntity<Set<Map<String, String>>> getTopics(
+      @PathVariable String bootstrapServers,
+      @Valid @PathVariable KafkaSupportedProtocol protocol,
+      @PathVariable String clusterName)
+      throws Exception {
+    Set<Map<String, String>> topics =
+        apacheKafkaTopicService.loadTopics(bootstrapServers, protocol, clusterName);
+    return new ResponseEntity<>(topics, HttpStatus.OK);
+  }
+
+  @RequestMapping(
+      value =
+          "/getAcls/{bootstrapServers}/{aclsNativeType}/{protocol}/{clusterName}/{projectName}/{serviceName}",
+      method = RequestMethod.GET,
+      produces = {MediaType.APPLICATION_JSON_VALUE})
+  public ResponseEntity<Set<Map<String, String>>> getAcls(
+      @PathVariable String bootstrapServers,
+      @Valid @PathVariable KafkaSupportedProtocol protocol,
+      @PathVariable String clusterName,
+      @PathVariable String aclsNativeType,
+      @PathVariable String projectName,
+      @PathVariable String serviceName)
+      throws Exception {
+    Set<Map<String, String>> acls;
+    if (AclsNativeType.NATIVE.name().equals(aclsNativeType)) {
+      acls = apacheKafkaAclService.loadAcls(bootstrapServers, protocol, clusterName);
+    } else {
+      acls = aivenApiService.listAcls(projectName, serviceName);
+    }
+    return new ResponseEntity<>(acls, HttpStatus.OK);
```

Comment:
I guess we need a kind of separate issue to wrap every endpoint call process in `try` ... `catch` which in case of `Exception`, `Error` , `Throwable` should log it and send `NOT_OK` response

Reply:
Agree, updated ticket #173